tutorial/real_proj/github-org-analyzer.md

```diff
@@ -0,0 +1,244 @@
+# **Building a [GitHub Organization Analyzer](https://github.com/rajkumardongre/github-org-analyzer) in Fortran, using `http-client`** 🚀
+
+In this tutorial, we'll create a simple Fortran program that uses the [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) to retrieve and display all the repositories of the [`fortran-lang`](https://github.com/fortran-lang) organization. We'll use the [`http-client`](https://github.com/fortran-lang/http-client) and [`json-fortran`](https://github.com/jacobwilliams/json-fortran) libraries to make API requests and handle JSON responses.
+
+# **Prerequisite** 🚩
+
+Before building the GitHub Organization Analyzer library and running the program, you need to ensure that you have [`fpm`](https://fpm.fortran-lang.org/) (Fortran Package Manager) installed. Additionally, there is one dependency required for the [`http-client`](https://github.com/fortran-lang/http-client) library used in the project. Follow the steps below to set up your environment:
+
+### **Step 1: Install fpm**
+
+[`fpm`](https://fpm.fortran-lang.org/) is the Fortran Package Manager used for building and managing Fortran projects. If you do not currently have `fpm` installed, you can follow the installation instructions available on the official `fpm` repository to install version v0.9.0 or a more recent version. The installation guide can be found here: [Installation Guide](https://fpm.fortran-lang.org/install/index.html).
+
+
+### **Step 2: Install libcurl Development Headers**
+
+The `http-client` library, requires the libcurl development headers to be installed. On Ubuntu-based systems, you can install the required dependencies using the following command:
+
+```
+sudo apt install -y libcurl4-openssl-dev
+```
+
+This command will install the necessary development headers for libcurl, enabling the `http-client` library to make API requests to fetch data from the GitHub API.
+
+Once you have `fpm` installed and the required dependencies set up, you are ready to proceed with building and running the GitHub Organization Analyzer project.🙌
+
+# **Let's Start Building** 👨‍💻
+
+### **Step 1: Set up the Project**
+
+>**Note : This requires at least fpm v0.9.0.**
+
+1. Open your terminal or command prompt and create a new directory for the project:
+
+```
+fpm new github-org-analyzer
+cd github-org-analyzer
+```
+
+2. The project structure will look like this:
+
+```
+.
+├── README.md
+├── app
+│   └── main.f90
+├── fpm.toml
+├── src
+│   └── github-org-analyzer.f90
+└── test
+    └── check.f90
+```
+
+### **Step 2: Add Dependencies to `fpm.toml`**
+
+Open the `fpm.toml` file and add the following dependencies:
+
+```toml
+[dependencies]
+http = { git = "https://github.com/fortran-lang/http-client.git" }
+stdlib = "*"
+json-fortran = { git = "https://github.com/jacobwilliams/json-fortran.git" }
+```
+
+### **Step 3: Build the Project**
+
+Run the following command to build the project:
+
+```
+fpm build
+```
```

Comment:
Is this step really needed here?

Reply:
In my understanding, when executing the `fpm run` command, the project will be built automatically during that process. Therefore, it seems unnecessary to include a separate step for building the project. So Should I remove this step?